core/src/scene/sceneLoader.cpp

```diff
@@ -147,12 +105,23 @@ void SceneLoader::loadShaderConfig(Node shaders, ShaderProgram& shader) {
         }
     }
 
-    Node uniformsNode = shaders["uniforms"];
-    if (uniformsNode) {
+    if (Node uniformsNode = shaders["uniforms"]) {
         for (const auto& uniform : uniformsNode) {
-            std::string name = uniform.first.as<std::string>();
-            std::string value = uniform.first.as<std::string>();
-            shader.addSourceBlock("uniforms", "uniform " + name + " = " + value + ";");
+            auto name = uniform.first.as<std::string>();
+            auto uniforms = parseStyleUniforms(uniform.second, scene);
+            auto& type = uniforms.first;
+            auto& uniformValues = uniforms.second;
+            int size = uniformValues.size();
+            if (size == 1) {
+                shader.addSourceBlock("uniforms", "uniform " + type + " " + name + ";");
```

Comment:
I put them all in `globals`, but there is no real reason we can't have a `uniforms` block instead, and doing that in JS may also help to more clearly segment things too.